src/ucp/proto/proto_common.inl

```diff
@@ -46,6 +46,11 @@ ucp_proto_bcopy_send_func_status(ssize_t packed_size)
 static UCS_F_ALWAYS_INLINE void
 ucp_proto_msg_multi_request_init(ucp_request_t *req)
 {
+    if (ucp_proto_select_op_flags(&req->send.proto_config->select_param) &
+        UCP_PROTO_SELECT_OP_FLAG_RESUME) {
+        return;
+    }
```

Comment:
won't it cause ordering issues?

Reply:
AFAIU it will prevent existing requests from getting a new message id.
new requests will not be effected.

Reply:
got it, maybe add an assert to make sure that `req->send.msg_proto.message_id` is initialized?

Reply:
check ucp_datatype_iter_is_begin